homeassistant/components/intent/__init__.py

```diff
@@ -89,10 +101,22 @@ async def async_setup_intents(self, hass: HomeAssistant) -> None:
 class OnOffIntentHandler(intent.ServiceIntentHandler):
     """Intent handler for on/off that also supports covers, valves, locks, etc."""
 
+    def __init__(
+        self,
+        intent_type: str,
+        domain: str,
+        service: str,
+    ) -> None:
+        """Create on/off intent handler."""
+        super().__init__(intent_type, domain, service)
+
     async def async_call_service(self, intent_obj: intent.Intent, state: State) -> None:
         """Call service on entity with handling for special cases."""
         hass = intent_obj.hass
 
+        # Required to be a str in constructor
+        assert self.service is not None
```

Comment:
Can you override the type on the attribute so this is not needed?

Reply:
Actually, it's weird it was changed in the base class. Why not just allow `None` in `SetPositionIntentHandler` ?

Reply:
This is what I did above in the constructor, which is why I have the assert here since it should cause a type checking error to pass `None` for service to this class.

Reply:
the other way around, that only for SetPosition it can be None? It feels like the class should be split up if we're not using that part.